app.py

```diff
@@ -267,18 +272,32 @@ def parsing_message(sender_id , message):
             msg = "Hey {} ! I don't know much but you can find more about GSoC(Google Summer of Code) at https://wiki.metakgp.org/w/Google_Summer_of_Code ".format(user_details['first_name'])
         except KeyError:
             msg = "I don't know much but you can find more about GSoC(Google Summer of Code) at https://wiki.metakgp.org/w/Google_Summer_of_Code "
-        send_message(sender_id, msg)
+
     elif dc_re_1 and dc_re_2 and dc_re_3 :
         hub_address = get_hub_add()
         try :
             msg = "Hi {} ! The current hub address is {}".format(user_details['first_name'],hub_address)
         except KeyError: 
             msg = "The current hub address is {}".format(hub_address)
-        send_message(sender_id, msg)
-    else :
+
+    elif learn_re_1:
+        if any(re.search(keyword, message, re.IGNORECASE)) for keyword in languages:
```

Comment:
I suggest we move our focus to `api.ai` first before adding more of these features because after that these kind of things will be a lot easier and more useful .